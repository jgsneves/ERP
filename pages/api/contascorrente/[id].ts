import { NextApiRequest, NextApiResponse } from "next";
import prisma from "../../../services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const { id } = req.query;

        const result = await prisma.contasCorrente.findUnique({
          where: {
            Id: id as string,
          },
        });

        if (result) {
          res.json(result);
          return;
        }

        res.status(404).send({ message: "Conta corrente não encontrada." });
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    case "DELETE":
      try {
        const { id } = req.query;

        const result = await prisma.contasCorrente.delete({
          where: {
            Id: id as string,
          },
        });

        res.json(result);
      } catch (error) {
        res.status(500).send({ error });
      }

      break;
    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade",
      });
      break;
  }
}
