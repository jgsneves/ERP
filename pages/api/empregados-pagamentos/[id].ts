import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "PATCH":
      try {
        const result = await prisma.empregadosPagamentos.update({
          data: req.body,
          where: {
            Id: req.query.id as string,
          },
        });
        res.json(result);
      } catch (error) {
        console.log({ error });
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
