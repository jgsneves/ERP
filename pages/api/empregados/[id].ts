import { NextApiRequest, NextApiResponse } from "next";
import prisma from "../../../services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.pessoas.findUnique({
          where: {
            Id: req.query.id as string,
          },
          include: {
            ContasCorrentes: true,
            Endereco: true,
          },
        });

        if (result) return res.json(result);

        res.status(404).send({ message: "Empregado não encontrado!" });
      } catch (error) {
        console.log({ error });

        res.status(500).send({ error });
      }
      break;

    case "PATCH":
      try {
        await prisma.pessoas.update({
          where: {
            Id: req.query.id as string,
          },
          data: req.body,
        });
        res.status(200).send({ message: "Empregado atualizado com sucesso! " });
      } catch (error) {
        res
          .status(500)
          .send({ message: "Não foi possível editar este empregado!" });
      }
      break;

    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
