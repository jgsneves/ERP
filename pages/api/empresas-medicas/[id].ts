import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.empresasMedicas.findUnique({
          where: {
            Id: req.query.id as string,
          },
          include: {
            Socios: true,
          },
        });

        if (result) return res.json(result);

        res.status(404).send({ message: "Empresa médica não encontrada! " });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
