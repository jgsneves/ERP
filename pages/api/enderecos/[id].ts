import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const { id } = req.query;
        const result = await prisma.enderecos.findUnique({
          where: {
            Id: id as string,
          },
        });
        if (result) {
          res.json(result);
          return;
        }
        res.status(404).send({ message: "Endereço não encontrado." });
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    case "PUT":
      try {
        const { body, query } = req;
        const result = await prisma.enderecos.update({
          where: {
            Id: query.id as string,
          },
          data: body,
        });
        res.status(200).send(result);
      } catch (error) {
        res.status(500).send({ error });
      }
      break;

    default:
      res.status(500).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
