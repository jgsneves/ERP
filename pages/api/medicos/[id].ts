import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const result = await prisma.pessoas.findUnique({
          where: {
            Id: req.query.id as string,
          },
        });

        if (result) {
          return res.json(result);
        }

        res.status(404).send({ message: "Médico não encontrado." });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    case "PATCH":
      try {
        const requestBody = req.body;

        await prisma.pessoas.update({
          where: {
            Id: req.query.id as string,
          },
          data: requestBody,
        });

        res.status(200).send({ message: "Médico atualizado com sucesso!" });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;
    default:
      res.status(400).send({
        method: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
