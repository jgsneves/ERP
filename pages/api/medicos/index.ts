import { Pessoas, PessoasTipo } from "@prisma/client";
import { NextApiRequest, NextApiResponse } from "next";
import prisma from "services/Prisma";
import { ErrorHandler } from "utils/ErrorHandler";

interface EmpresaMedicaRazaoSocial {
  RazaoSocial: string;
}

interface Doctors extends Pessoas {
  EmpresaMedica: EmpresaMedicaRazaoSocial | null;
}

export interface MedicosResponse {
  pagina: number;
  totalPaginas: number;
  doctors: Doctors[];
}

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  switch (req.method) {
    case "GET":
      try {
        const pagina = parseInt(req.query.pagina as string) || 1;
        const quantidade = parseInt(req.query.quantidade as string) || 10;
        const empresaMedicaId = req.query.empresaMedicaId;

        const whereClause = {
          EmpresaMedicaId:
            typeof empresaMedicaId === "string" ? empresaMedicaId : undefined,
          Tipo: PessoasTipo.MEDICO,
        };

        const doctors = await prisma.pessoas.findMany({
          skip: (pagina - 1) * quantidade,
          take: quantidade,
          orderBy: {
            Nome: "asc",
          },
          where: whereClause,
          include: {
            EmpresaMedica: {
              select: {
                RazaoSocial: true,
              },
            },
          },
        });

        const count = doctors.length;
        const totalPaginas = Math.ceil(count / quantidade);
        res.json({ pagina, totalPaginas, doctors });
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;

    case "POST":
      try {
        const result = await prisma.pessoas.create({
          data: req.body,
        });
        res.json(result);
      } catch (error) {
        ErrorHandler.logPrismaError(error);
        res.status(500).send({ error });
      }
      break;
    default:
      res.status(400).send({
        metodo: req.method,
        message: "Método não implementado para esta entidade!",
      });
      break;
  }
}
