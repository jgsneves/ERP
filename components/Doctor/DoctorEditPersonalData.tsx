import { formatCPF } from "@brazilian-utils/brazilian-utils";
import {
  VStack,
  FormControl,
  FormLabel,
  Input,
  Select,
  Button,
  Flex,
  useToast,
} from "@chakra-ui/react";
import { EstadoCivil, Pessoas } from "@prisma/client";
import axios from "axios";
import ContentTitle from "components/Shared/ContentTitle";
import { useRouter } from "next/router";
import { DoctorEntity } from "pages/medicos/[id]";
import { Dispatch, SetStateAction, useState } from "react";
import { CellphoneNumberFormat } from "utils/CellphoneNumberFormat";
import { DateFormat } from "utils/DateFormat";
import { ErrorHandler } from "utils/ErrorHandler";

interface Props {
  doctor: DoctorEntity;
  setIsEditState: Dispatch<SetStateAction<boolean>>;
}

export default function DoctorEditPersonalData({
  doctor,
  setIsEditState,
}: Props) {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [dateInputValue, setDateInputValue] = useState<string>(
    DateFormat.getChakraDateFormat(new Date(doctor.DataNascimento))
  );
  const [formData, setFormData] = useState<Pessoas>({
    Cpf: doctor.Cpf,
    Email: doctor.Email,
    EstadoCivil: doctor.EstadoCivil,
    Nacionalidade: doctor.Nacionalidade,
    PIS: doctor.PIS,
    RG: doctor.RG,
    TelefonePessoal: doctor.TelefonePessoal,
    TituloEleitor: doctor.TituloEleitor,
    CriadoEm: doctor.CriadoEm,
    Crm: doctor.Crm,
    DataNascimento: new Date(doctor.DataNascimento),
    EmpresaMedicaId: doctor.EmpresaMedicaId,
    EnderecoId: doctor.EnderecoId,
    Id: doctor.Id,
    ModificadoEm: new Date(),
    Nome: doctor.Nome,
    Participacao: null,
    Salario: null,
    StatusAdmissao: null,
    Tipo: "MEDICO",
    ModalidadeTrabalho: null,
  });

  const toast = useToast();
  const router = useRouter();

  const handleInputOnChange = (
    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    const { id, value } = event.target;
    setFormData((state) => ({ ...state, [id]: value }));
  };

  const handleDateOnBlur = () => {
    if (dateInputValue) {
      const date = DateFormat.getDateTypeFromChakraString(dateInputValue);
      setFormData((state) => ({ ...state, DataNascimento: date }));
    }
  };

  const handleSubmitOnClick = () => {
    setIsLoading(true);

    axios
      .patch(`/api/medicos/${doctor.Id}`, formData)
      .then(
        () => {
          toast({
            title: "Médico editado com sucesso!",
            duration: 5000,
            status: "success",
          });
        },
        () => {
          toast({
            title: "Não foi editar os dados do médico!",
            duration: 9000,
            isClosable: true,
            status: "error",
          });
        }
      )
      .catch((error) => ErrorHandler.logAxiosPatchError(error))
      .finally(() => {
        setIsLoading(false);
        router.reload();
      });
  };

  return (
    <VStack alignItems="flex-start">
      <ContentTitle title="Dados pessoais" />
      <FormControl maxW="500px">
        <FormLabel>
          Nome
          <Input
            isDisabled={isLoading}
            value={formData.Nome}
            id="Nome"
            onChange={handleInputOnChange}
          />
        </FormLabel>
        <FormLabel>
          Telefone pessoal:
          <Input
            value={CellphoneNumberFormat.format(formData.TelefonePessoal)}
            id="TelefonePessoal"
            onChange={handleInputOnChange}
          />
        </FormLabel>

        <FormLabel>
          E-mail:
          <Input
            value={formData.Email}
            id="Email"
            onChange={handleInputOnChange}
            type="email"
          />
        </FormLabel>

        <FormLabel>
          CPF
          <Input
            isDisabled={isLoading}
            value={formatCPF(formData.Cpf)}
            id="Cpf"
            onChange={handleInputOnChange}
          />
        </FormLabel>
        <FormLabel>
          CRM
          <Input
            isDisabled={isLoading}
            value={formData.Crm ?? ""}
            id="Crm"
            onChange={handleInputOnChange}
          />
        </FormLabel>
        <FormLabel>
          RG:
          <Input value={formData.RG} id="RG" onChange={handleInputOnChange} />
        </FormLabel>

        <FormLabel>
          PIS:
          <Input
            value={formData.PIS ?? ""}
            id="PIS"
            onChange={handleInputOnChange}
          />
        </FormLabel>
        <FormLabel>
          Date de nascimento
          <Input
            isDisabled={isLoading}
            type="date"
            value={dateInputValue}
            onChange={(event) => setDateInputValue(event.target.value)}
            onBlur={handleDateOnBlur}
          />
        </FormLabel>
        <FormLabel>
          Nacionalidade:
          <Input
            value={formData.Nacionalidade}
            id="Nacionalidade"
            onChange={handleInputOnChange}
          />
        </FormLabel>

        <FormLabel>
          Estado civil:
          <Select
            value={formData.EstadoCivil}
            onChange={handleInputOnChange}
            id="EstadoCivil"
          >
            {Object.values(EstadoCivil).map((estadoCivil) => (
              <option key={estadoCivil} value={estadoCivil}>
                {estadoCivil}
              </option>
            ))}
          </Select>
        </FormLabel>
        <Flex alignItems="center" mt={2} gap={3}>
          <Button
            isLoading={isLoading}
            colorScheme="green"
            onClick={handleSubmitOnClick}
          >
            salvar
          </Button>
          <Button
            variant="ghost"
            colorScheme="green"
            onClick={() => setIsEditState(false)}
          >
            cancelar
          </Button>
        </Flex>
      </FormControl>
    </VStack>
  );
}
