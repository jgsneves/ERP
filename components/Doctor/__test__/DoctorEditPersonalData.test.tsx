import { render } from "@testing-library/react";
import { DoctorEntity } from "pages/medicos/[id]";
import DoctorEditPersonalData from "../DoctorEditPersonalData";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("DoctorEditPersonalData", () => {
  const mockedDoctorData: DoctorEntity = {
    Cpf: "022.022.022-22",
    CriadoEm: new Date(),
    Crm: "4645",
    DataNascimento: "1999-31-31",
    Email: "teste@gmail.com",
    EmpresaMedicaId: null,
    EnderecoId: null,
    EstadoCivil: "CASADO",
    Id: "1",
    Nacionalidade: "brasileiro",
    Nome: "JosÃ©",
    RG: "11111",
    TelefonePessoal: "(71) 9999-999",
    ModificadoEm: null,
    PIS: null,
    TituloEleitor: null,
  };
  it("should rende proper snapshot", () => {
    const { asFragment } = render(
      <DoctorEditPersonalData
        doctor={mockedDoctorData}
        setIsEditState={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
