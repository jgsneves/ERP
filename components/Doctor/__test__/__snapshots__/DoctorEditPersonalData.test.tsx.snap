// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorEditPersonalData should rende proper snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1h4ws66"
  >
    <div
      class="css-8v4id7"
    >
      <b
        class="chakra-text css-r262sy"
      >
        Dados pessoais
      </b>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
    </div>
    <div
      class="chakra-form-control css-1cl7zdz"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Nome
        <input
          class="chakra-input css-0"
          id="Nome"
          value="JosÃ©"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Telefone pessoal:
        <input
          class="chakra-input css-0"
          id="TelefonePessoal"
          value="(71) 999-9999"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        E-mail:
        <input
          class="chakra-input css-0"
          id="Email"
          type="email"
          value="teste@gmail.com"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        CPF
        <input
          class="chakra-input css-0"
          id="Cpf"
          value="022.022.022-22"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        CRM
        <input
          class="chakra-input css-0"
          id="Crm"
          value="4645"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        RG:
        <input
          class="chakra-input css-0"
          id="RG"
          value="11111"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        PIS:
        <input
          class="chakra-input css-0"
          id="PIS"
          value=""
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Date de nascimento
        <input
          class="chakra-input css-0"
          id="field-:r0:"
          type="date"
          value="Invalid Date-Invalid Date-Invalid Date"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Nacionalidade:
        <input
          class="chakra-input css-0"
          id="Nacionalidade"
          value="brasileiro"
        />
      </label>
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Estado civil:
        <div
          class="chakra-select__wrapper css-42b2qy"
        >
          <select
            class="chakra-select css-1ik61og"
            id="EstadoCivil"
          >
            <option
              value="SOLTEIRO"
            >
              SOLTEIRO
            </option>
            <option
              value="CASADO"
            >
              CASADO
            </option>
            <option
              value="SEPARADO"
            >
              SEPARADO
            </option>
            <option
              value="DIVORCIADO"
            >
              DIVORCIADO
            </option>
            <option
              value="VIUVO"
            >
              VIUVO
            </option>
          </select>
          <div
            class="chakra-select__icon-wrapper css-162mkon"
          >
            <svg
              aria-hidden="true"
              class="chakra-select__icon"
              focusable="false"
              role="presentation"
              style="width: 1em; height: 1em; color: currentColor;"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </label>
      <div
        class="css-tff4n3"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          salvar
        </button>
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          cancelar
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
